### PR TITLE
LL-6472 Sticky firmware update banner

### PR DESCRIPTION
**Pull request by juan-cortes**
This pr adds a sticky banner that will be visible throughout the app to funnel users to update their firmware if the _last seen device_ that accessed the manager had an update available. We don't check again until the user accesses the manager so perhaps this is something to improve in the future. 
- When both the app update banner and the firmware update banner would be visible we will show **only** the app update banner
- Both banners are now orange, no blue banners.
- I changed the error banner to red when the download fails, to match the dot on the sidebar

### Type

Feature

### Context

https://ledgerhq.atlassian.net/browse/LL-6472

### Screenshots
image
image
image

### Parts of the app affected / Test plan

This needs to be tested from several angles, I've added a few flags to ease testing. In particular, a new debug menu triggered by `DEBUG_FIRMWARE_UPDATE=1` will allow you to see the latest known device that would be used, and the fw version available if any. You can also override the version to trigger the banner by passing a `DEBUG_FW_VERSION=123` where `123` is the version you want to be displayed on the app (if inside the manager, it will only work if the device has an update available). 
To be sure no flows are broken, test with:

- A device that's in the latest fw version available, should not show the banner.
- A device that's in an updatable fw version, should show the banner.
- A device that's in a version that's too old to be updated, should show a different banner inside the manager, pointing to support for a replacement. The CTA should open the support URL on the browser.
- While installing/uninstalling apps, the CTA in the banner should look disabled (partial opacity)

-  Clicking on the "Go to manager" CTA on the banner should open the manager screen with the update firmware modal open by default, the user should be able to close this banner.
- Clicking on "Update firmware" while in the manager, should open the modal for the update firmware flow.

### CODE REVIEW

**Comment by juan-cortes** [comment]
Updated all the screenshots that showed the old update firmware banner in favor of the new orange one, hopefully this is enough for spectron

**Comment by juan-cortes** [comment]
So @cthepot-ledger, your call on how this should look if it's not what you expected.

**Comment by cthepot-ledger** [comment]
@juan-cortes change of scope, apologies

1. can we ensure the FW update banner only displayed when Live is on the latest version -- the banner does not show when Live is outdated
2. can we change the colour of the alert of the Live update banner from blue to orange

**Comment by juan-cortes** [comment]
> @juan-cortes change of scope, apologies
> 
> 1. can we ensure the FW update banner only displayed when Live is on the latest version -- the banner does not show when Live is outdated
> 2. can we change the colour of the alert of the Live update banner from blue to orange

@cthepot-ledger  So only one is visible at a time with priority for the app update over the firmware update, both are orange.
Does this mean that even in the manager I would see the app update over the firmware update? And I can ask again, both look orange, none blue?

**Comment by cthepot-ledger** [comment]
@juan-cortes 
- both do look orange and not blue.
- also they should never be displayed at the same time in any case. because the FW banner is only displayed when the Live is on the latest version
hope it clarifies

**Comment by juan-cortes** [comment]
Crystal clear 👌🏼

**Comment by juan-cortes** [comment]
Will need to update the screenshots again once they are generated

**Comment by porenes-ledger** [comment]
@juan-cortes can you rename the PR, it's ticket LL-6472, not LL-6473 ;-) Thanks !

File: src/renderer/components/FirmwareUpdateBanner.js
Comment (jgiamporcaro-ledger) [inline src/renderer/components/FirmwareUpdateBanner.js:41]: --This message is just a suggestion and not a blocking comment--

For stuff like that, we can use `context` from the i18n library, the feature fit perfectly with this kind of behaviour:

Documentation: https://www.i18next.com/translation-function/context#context

---

JSON file
```json
{
      "warning": "...",
      "warning_old": "...",
} 
```

JS file
```jsx
<Trans i18nKey="manager.firmware.banner.warning" tOptions={{context: old ? "old" : null}} values={{ latestFirmware: visibleFirmwareVersion }} />
```

It's clearer form the translation file that both keys are paired.

File: src/renderer/screens/manager/FirmwareUpdate/index.js
Comment (jgiamporcaro-ledger) [inline src/renderer/screens/manager/FirmwareUpdate/index.js:124]: Good to know, curly braces are no longer required when dealing with string props. From my mind they add some noise.

Comment (jgiamporcaro-ledger) [inline src/renderer/screens/manager/FirmwareUpdate/index.js:124]: --This message is just a suggestion and not a blocking comment--

Good to know curly braces are no longer required when dealing with strings. Personally, I prefer to avoid it to reduce noise added to the JSX.

File: src/renderer/components/FirmwareUpdateBanner.js
Comment (juan-cortes) [reply src/renderer/components/FirmwareUpdateBanner.js:41]: Feel free to commit this to the pr, looks clearer to me too

File: src/renderer/components/FirmwareUpdateBanner.js
Comment (juan-cortes) [reply src/renderer/components/FirmwareUpdateBanner.js:36]: I can't make code contributions right now, but feel free, as long as the logic remains the same

File: src/renderer/screens/manager/FirmwareUpdate/index.js
Comment (juan-cortes) [reply src/renderer/screens/manager/FirmwareUpdate/index.js:124]: Old habits die hard, I suggest you raise the concern to the core team and ask them to add something like https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/jsx-curly-brace-presence.md to our eslint so that it's handled automatically

**Review by gre — COMMENTED** [review]
merge of develop needed, sorry folks

File: src/renderer/actions/settings.js
Comment (juan-cortes) [reply src/renderer/actions/settings.js:156]: You tell me, I don't have a better typing

**Comment by juan-cortes** [comment]
Merged, although I'm sure it will have to regenerate all the changed screenshots again. So lets wait for ci

**Review by LFBarreto — DISMISSED** [review]

**Comment by LFBarreto** [comment]
@juan-cortes can you regenerate screenshots. thanks :pray:

**Comment by juan-cortes** [comment]
I'm a need a merge from develop too considering it's 22 days old, I'll get to it as soon as I'm home

**Review by ychen-ledger — CHANGES_REQUESTED** [review]
Hi @juan-cortes , please find below, in the manger page, we still have both leger live update banner and Firmware update banner: 
Screenshot 2021-08-24 at 18 04 15

**Review by LFBarreto — DISMISSED** [review]

**Review by ychen-ledger — APPROVED** [review]
Validated with: 

- Modified nano package.json to ledger live 2.0.0
- FW 1.2.4-6
- Debug mode to perform download in progress/download success/download failed
- MAxOS Big sur 11.4

**Comment by juan-cortes** [comment]
Merged develop and updated screenshots again, lets see if this time is the charm

**Review by LFBarreto — APPROVED** [review]

**Review by valpinkman — APPROVED** [review]

**Pull request merged**
